### PR TITLE
[Idea] Add example about running a minimal web server

### PR DESCRIPTION
**Pull request by wilzbach**
Marking this "needs work" as I would need to remove `RUNABLE_EXAMPLE` from the `EXTRA_EXAMPLE` macro.

### CODE REVIEW

**Comment by MartinNowak** [comment]
Even vibe.d's front page finally switched to a normal main function ;). http://vibed.org/
```d
#!/usr/bin/env dub
/+ dub.sdl:
 name "hello_vibed"
 dependency "vibe-d" version="~>0.8.0-rc.2"
 +/
import vibe.vibe;

void main()
{
    auto settings = new HTTPServerSettings;
    settings.port = 8080;

    listenHTTP(settings, (req, res) { res.writeBody("Hello, World: " ~ req.path); });
    runApplication();
}
```

**Comment by wilzbach** [comment]
@CyberShadow - what's your opinion on this? It wouldn't be hard to hide the run buttons and maybe show additional explanation and links if we want to go this road.
I am not really decided here,
+ Vibe.d is the defacto flagship library
+ We want to show D code for different areas
+ It shows how awesome DUB's single file descriptions are

- it's not runnable
- it's a third-party library and not "official" D

**Comment by CyberShadow** [comment]
I think it's still worth adding.

When I asked an IRL colleague whether he'd be interested in using D, the first reason he said was that D "doesn't even have a web server in its standard library!". Starting up a web server is one of the more common front page examples of other languages, so I think we shouldn't stay behind here.

Lack of dub support on the example runner is just a lucky coincidence, since it's not meaningful to start a web server there either :)

**Comment by wilzbach** [comment]
Rebased the example to the option chooser example.

- `runApplication` isn't `@safe`. With my initial `VibeDefaultMain`,  `shared static this() @safe` and would work. What option do you prefer?
- Should we use `#!/usr/bin/env rdmd` for the other examples?

**Comment by wilzbach** [comment]
Oh and for convenience:

image

**Comment by CyberShadow** [comment]
> What option do you prefer?

`void main()` IMO

> Should we use #!/usr/bin/env rdmd for the other examples?

I think there's no need for that - vertical space is at a premium anyhow

File: index.dd [hunk 1]
```diff
@@ -94,6 +94,25 @@ void main()
 }
 ----
 )
+$(EXTRA_EXAMPLE Start a minimal web server,
+----
+#!/usr/bin/env dub
+/+ dub.sdl:
+name "hello_vibed"
+dependency "vibe-d" version="~>0.8.0"
```
Comment (wilzbach) [inline index.dd]: Vibe.d 0.8.0 will be released very [soon](http://forum.dlang.org/post/oj198q$1j9q$1@digitalmars.com), so I already accounted for the approval / review lag.

File: index.dd [hunk 2]
```diff
@@ -94,6 +94,25 @@ void main()
 }
 ----
 )
+$(EXTRA_EXAMPLE Start a minimal web server,
+----
+#!/usr/bin/env dub
+/+ dub.sdl:
+name "hello_vibed"
```
Comment (wilzbach) [inline index.dd]: FYI: With newer DUB instances this line won't be needed anymore (-> https://github.com/dlang/dub/pull/1081), but for now we have to wait a couple of months until https://github.com/dlang/dub/pull/1081 has gone through a couple of releases.

**Comment by CyberShadow** [comment]
@wilzbach Is this good to go? I see it still has the "needs work" label.

**Comment by CyberShadow** [comment]
Oh, or is it blocked by the Vibe.d release?

**Comment by wilzbach** [comment]
> Oh, or is it blocked by the Vibe.d release?

Yes, it's in three days. I think we can wait until then. I will ping this PR then ;-)

**Comment by wilzbach** [comment]
> Yes, it's in three days. I think we can wait until then. I will ping this PR then ;-)

Finally Vibe.d 0.8 was relased :tada:

http://forum.dlang.org/post/mmfrsonxrfxwltkfvliw@forum.dlang.org

**Comment by CyberShadow** [comment]
Darn it, doesn't work on my machine because of the OpenSSL issue.

Do you think we should go ahead with this despite this problem, or wait until Vibe's OpenSSL bindings are updated?

**Comment by wilzbach** [comment]
> Darn it, doesn't work on my machine because of the OpenSSL issue.

Hehe that's a well-known problem, see e.g. http://forum.dlang.org/post/bpjzqbdhwzplqrvsjfca@forum.dlang.org

In short, the workarounds are:
- `versions "VibeNoSSL"`
-  [setting libs-posix to openssl1.0](https://github.com/rejectedsoftware/vibe.d/issues/1748#issuecomment-297647832)
-  Use the botan library (pure D) with:
```
dependency "vibe-d:tls" version="~>0.8.0"
subConfiguration "vibe-d:tls" "botan"
```
This was a last minute change to 0.8.0: https://github.com/rejectedsoftware/vibe.d/pull/1823

The [proper fix](https://github.com/rejectedsoftware/vibe.d/pull/1759) has still problems due to DUB's inability to pass the linked library version.

> Do you think we should go ahead with this despite this problem, or wait until Vibe's OpenSSL bindings are updated?

- Sönke merged a similar PR to the README last week: https://github.com/rejectedsoftware/vibe.d/pull/1803
- A similar example is on [vibed.org](http://vibed.org)

-> Yeah - I think Arch users are a minority and they are used to breakage ;-)

**Pull request merged**